src/textual/reactive.py

```diff
@@ -50,6 +53,108 @@ class TooManyComputesError(ReactiveError):
     """Raised when an attribute has public and private compute methods."""
 
 
+class WatchDecorator(Generic[WatchMethodType]):
```

Comment:
This (and the `Validate`/`ComputeDecorator`) kind of reminded me of the typing struggles I had with the work decorator.
Maybe you did it already but it may be worth running `make typecheck` and making sure this introduces as little typing errors as possible.

Reply:
There is one, because I couldn't figure out how to type an unbound method. Any suggestions?

Reply:
I think I do have a suggestion.
The decorators (Watch / Compute / Validate) shouldn't be generic over the type of the method they're decorating; rather, they should be generic over the type of the reactive involved because that's what can really change.

For example, for my compute methods, I know I'll always be decorating methods with type `Callable[[Any], ReactiveType]`.
So, what _maybe_ I can have is an alias for compute methods and then use that.

The suggestion below seems to eliminate an error you get with `make typecheck` and it also gets rid of some IDE errors I get in several places.
However, I'm not 100% sure this is better.
It _feels_ better, but I wouldn't bet my life on it.

```py
# ...

ComputeMethodAlias: TypeAlias = Callable[[Any], ReactiveType]  # <--

# ...

class ComputeDecorator(Generic[ReactiveType]):  # <--
    def __init__(self, reactive: Reactive[ReactiveType] | None = None) -> None:  # <-- better typing here
        self._reactive = reactive

    @overload
    def __call__(self) -> ComputeDecorator[ReactiveType]: ...  # <--

    @overload
    def __call__(
        self, method: ComputeMethodAlias[ReactiveType]  # <--
    ) -> ComputeMethodAlias[ReactiveType]: ...  # <--

    def __call__(
        self, method: ComputeMethodAlias[ReactiveType] | None = None  # <--
    ) -> ComputeMethodAlias[ReactiveType] | ComputeDecorator[ReactiveType]:  # <--
        # ...

# ...

@rich.repr.auto
class Reactive(Generic[ReactiveType]):
    # ...
    def __init__(...):
        # ...
        self._compute_method: ComputeMethodAlias[ReactiveType] | None = None  # <--  better typing

    # ...

    @property
    def compute(self) -> ComputeDecorator[ReactiveType]:  # <--
        # ...
```